src/shogun/base/SGObject.h

```diff
@@ -278,6 +281,116 @@ class CSGObject
 	 */
 	void build_gradient_parameter_dictionary(CMap<TParameter*, CSGObject*>* dict);
 
+	/** Checks if object has a class parameter identified by a name.
+	 *
+	 * @param name name of the parameter
+	 * @return true if the parameter exists with the input name
+	 */
+	bool has(const std::string& name) const
+	{
+		BaseTag tag(name);
+		return has_with_base_tag(tag);
+	}
+
+	/** Checks if object has a class parameter identified by a Tag.
+	 *
+	 * @param tag tag of the parameter containing name and type information
+	 * @return true if the parameter exists with the input tag
+	 */
+	template <typename T>
+	bool has(const Tag<T>& tag) const
+	{
+		return has<T>(tag.name());
+	}
+
+	/** Checks if a type exists for a class parameter identified by a name.
+	 *
+	 * @param name name of the parameter
+	 * @return true if the parameter exists with the input name and type
+	 */
+	template <typename T, typename U=void>
```

Comment:
This is for SWIG interface, to differentiate between `has<T>(string)` and `has<T>(Tag)`. I had mentioned this in the docs earlier but heiko made me remove this since these docs are visible to shogun users, and they don't know what SWIG is. Guess I should add this in the readme.